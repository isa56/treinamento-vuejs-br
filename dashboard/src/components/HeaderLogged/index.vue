<script>
import { computed } from "vue"
import { useRouter } from "vue-router"

import useStore from "@/hooks/useStore"
import { cleanCurrentUser } from "@/store/user"

export default {
    setup() {
        const store = useStore()
        const router = useRouter('User')

        console.log("Current user", store.currentUser)
        console.log("Current user name", store.currentUser.name)

        const logoutLabel = computed(() => {
            if (store.currentUser.name) {
                return '...'
            }
            return `${store.currentUser.name}, sair`
        })

        function handleLogout() {
            window.localStorage.removeItem('token')
            cleanCurrentUser
            router.push({name: 'home'})
        }

        return {
            router,
            logoutLabel,
            handleLogout
        }
    },

}
</script>


<template>
    <div class="flex items-center justify-between w-4/5 max-w-6xl py-10">
        <div class="w-28 lg:w-36">
            <img src="@/assets/images/logo_white.png" alt="Logo" class="w-full" />
        </div>
        <div class="flex">
            <ul class="flex list-none">
                <li
                    class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
                    @click="() => router.push({ name: 'credentials' })"
                >Credenciais</li>
                <li
                    class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none"
                    @click="() => router.push({ name: 'feedbacks' })"
                >Feedbacks</li>
                <li
                    class="px-5 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none"
                    @click="handleLogout"
                >{{ logoutLabel }}</li>
            </ul>
        </div>
    </div>
</template>


<style scoped>
</style>
