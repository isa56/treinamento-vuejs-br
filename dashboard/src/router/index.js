import { createRouter, createWebHistory } from 'vue-router'

const HomeView = import('../views/Home/index.vue')
const FeedbacksView = import('@/views/Feedbacks/index.vue')
const CredentialsView = import('../views/Credentials/index.vue')


const routes = [
  {
    path: '/',
    name: 'home',
    component: HomeView
  },
  {
    path: '/feedbacks',
    name: 'feedbacks',
    component: FeedbacksView,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/credentials',
    name: 'credentials',
    component: CredentialsView,
    meta: {
      hasAuth: true
    }
  },
  {
    path: '/:pathMatch(.*)*',
    redirect: { name: 'home' }
  }
]

const router = createRouter({
  history: createWebHistory('/'),
  routes
})

export default router