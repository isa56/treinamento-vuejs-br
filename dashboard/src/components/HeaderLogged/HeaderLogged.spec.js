import HeaderLogged from '.'

import { shallowMount } from '@vue/test-utils'
import { routes } from '@/router'
import { createRouter, createWebHistory } from 'vue-router'

const router = createRouter({
    history: createWebHistory('/'),
    routes
})

const mockStore = { currentUser: {} }

jest.mock(('@/hooks/useStore'), () => {
    return () => {
        return mockStore
    }
})


describe('<HeaderLogged />', () => {

    it('should render header logged correctly', async () => {

        router.push('/')
        await router.isReady()

        const wrapper = shallowMount(HeaderLogged, {
            global: {
                plugins: [router]
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    })

    it('should render 3 dots when there\'s no user is logged', async () => {

        router.push('/')
        await router.isReady()

        const wrapper = shallowMount(HeaderLogged, {
            global: {
                plugins: [router]
            }
        })

        const buttonLogout = wrapper.find('#logout-button')

        expect(buttonLogout.text()).toBe('...')

    })

    it('should render user name when there\'s a logged user', async () => {

        router.push('/')
        await router.isReady()

        mockStore.currentUser.name = 'Isa'
        const wrapper = shallowMount(HeaderLogged, {
            global: {
                plugins: [router]
            }
        })

        const buttonLogout = wrapper.find('#logout-button')

        expect(buttonLogout.text()).toBe('Isa, sair')


    })

})