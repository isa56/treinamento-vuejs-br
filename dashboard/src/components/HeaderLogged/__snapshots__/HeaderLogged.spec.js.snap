// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderLogged /> should render header logged correctly 1`] = `
<div class="flex items-center justify-between w-full max-w-6xl py-10">
  <div class="w-28 lg:w-36"><img src="" alt="Logo" class="w-full"></div>
  <div class="flex">
    <ul class="flex list-none">
      <li id="credentials-access-button" class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none">Credenciais</li>
      <li class="px-6 py-2 mr-2 font-bold text-white rounded-full cursor-pointer focus:outline-none">Feedbacks</li>
      <li class="px-5 py-2 font-bold bg-white rounded-full cursor-pointer text-brand-main focus:outline-none" id="logout-button">...</li>
    </ul>
  </div>
</div>
`;
